htdocs/projet/tasks/time.php

```diff
@@ -587,7 +587,12 @@
 
 					$arrayoftasks[$object->timespent_id]['timespent'] = $object->timespent_duration;
 					$arrayoftasks[$object->timespent_id]['totalvaluetodivideby3600'] = $object->timespent_duration * $object->timespent_thm;
-					$arrayoftasks[$object->timespent_id]['note'] = $ftask->ref.' - '.$ftask->label.' - '.$username.($object->timespent_note ? ' - '.$object->timespent_note : '');		// TODO Add user name in note
+					$arrayoftasks[$object->timespent_id]['note'] = $ftask->ref.' - '.$ftask->label.' - '.$username;		// TODO Add user name in note
+					if (!empty($conf->fckeditor->enabled) && !empty($conf->global->FCKEDITOR_ENABLE_DETAILS) && !empty($object->timespent_note)) {
```

Comment:
FCKEDITOR_ENABLE_DETAILS on or off is the current state. it may differs from the state of option when data was recorded into database.
For this reason, you must use 
dol_concatdesc() to concat strings without regarding value of FCKEDITOR_ENABLE_DETAILS and module.
so result will be good even if timespent note was recorded hen option differs of the task note.

Reply:
`timespent note` is never with rich text editor (for now).

The goal is to put the  `timespent note`  text after the line label.
With dol_concatdesc it put all on the same line and remove the line returns on `timespent note` 

`$arrayoftasks[$object->timespent_id]['note']=dol_concatdesc($arrayoftasks[$object->timespent_id]['note'],$object->timespent_note);`

will result with 
`TK1607-0004 - Project preparation phase A - Alice Adminson aaaa bbbb ccc`
instead of 
```
TK1607-0004 - Project preparation phase A - Alice Adminson
aaaa
bbbb
ccc
```

Reply:
No dol_concatdesc does not concat on same line.
It concats on a new line taking care for you if content is text or html.

Reply:
in this case there is a bug in dol_concatdesc
The new commit fix it